rustls/src/x509.rs

```diff
@@ -28,17 +29,37 @@ pub(crate) fn asn1_wrap(tag: u8, bytes: &[u8]) -> Vec<u8> {
         ret.push(0x80 + encoded_bytes as u8);
         ret.extend_from_slice(&size[leading_zero_bytes..]);
 
-        ret.extend_from_slice(bytes);
+        ret.extend_from_slice(bytes_a);
+        ret.extend_from_slice(bytes_b);
         ret
     }
 }
 
 /// Prepend stuff to `bytes` to put it in a DER SEQUENCE.
 pub(crate) fn wrap_in_sequence(bytes: &[u8]) -> Vec<u8> {
-    asn1_wrap(DER_SEQUENCE_TAG, bytes)
+    asn1_wrap(DER_SEQUENCE_TAG, bytes, &[])
+}
+
+/// Prepend stuff to `bytes_a` + `bytes_b` to put it in a DER SEQUENCE.
+#[cfg_attr(not(feature = "ring"), allow(dead_code))]
+pub(crate) fn wrap_concat_in_sequence(bytes_a: &[u8], bytes_b: &[u8]) -> Vec<u8> {
+    asn1_wrap(DER_SEQUENCE_TAG, bytes_a, bytes_b)
+}
+
+/// Prepend stuff to `bytes` to put it in a DER BIT STRING.
+pub(crate) fn wrap_in_bitstring(bytes: &[u8]) -> Vec<u8> {
+    asn1_wrap(DER_BITSTRING_TAG, &[0u8], bytes)
```

Comment:
Just passing by—feel free to disregard. I see that this creates an ASN.1 `BIT STRING` out of some bytes, but always encodes the length as zero instead of the number of bytes. Is this legal/intentional?

Reply:
The `BIT STRING` encoding starts with a single octet on the front saying how many bits to disregard from the last octet.  So this zero means "no bits" unused, which is correct because our input is an string of octets.

Reply:
Will add this comment into a test.

Reply:
Gotcha. Thanks for the insight!